virttest/nfs.py

```diff
@@ -332,6 +332,32 @@ def setup_remote(self):
                 raise exceptions.TestFail("Failed to run %s: %s" % (mkdir_cmd, o))
             self.mkdir_mount_remote = True
 
+        # Check firewall in host permits nfs service to mount nfs directory
+        # from remote machine
+        try:
+            firewall_cmd = "firewall-cmd --list-all | grep services:"
+            status, output = commands.getstatusoutput(firewall_cmd)
+            if not status:
+                firewall_services = output.split(':')[1].strip().split(' ')
+                if 'nfs' not in firewall_services:
+                    service_cmd = "firewall-cmd --permanent --zone=public "
+                    service_cmd += "--add-service=nfs && "
```

Comment:
look at virttest/staging/service.py maybe this module will help you reduce some code.